### PR TITLE
Origin/moore garrett welcome

### PR DESCRIPTION
**Pull request by gmoore10**
All requirements for Release Day 1 implemented.

### CODE REVIEW

**Comment by Kankur0** [comment]
Everything works fine, but I did fine two places that could improve the code.

Within the client.js file it is checking if the channel exists by using an if statement. This is good, but the code may cause an error since right after it tells the bot to send the message. It might be more ideal to put the ``channel.send`` method inside of an else statement.

#### Example
```
if (!channel) {
    util.log('Channel Not Found!', channel, 3);
} else {
    channel.send(`Welcome to the server ${member}!`);
};
```

The other place that might be a place to improve upon is in the welcome.js file. The method sends a message and then returns the string "done". This seems a bit unneeded. Instead we could think about having it just returning the message send method itself. 

##### Remove
``return 'Done.';``
##### Change
``return message.member.send(`Welcome to our server ${message.member}!`);``

Other than these two changes, I think the code looks great!

**Comment by gmoore10** [comment]
I've updated my code to reflect your requested changes. Thanks!

**Comment by Kankur0** [comment]
Looks good to me!

**Review by Kankur0 — APPROVED** [review]
Functions properly and abides by style guide.

File: package.json [hunk 1]
```diff
@@ -15,20 +15,25 @@
...
+    "chai": "^4.1.2",
+    "dependencies": "0.0.1",
     "discord.js": "^11.1.0",
     "dotenv": "^4.0.0",
+    "erlpack": "github:discordapp/erlpack",
     "eslint": "^4.3.1",
     "eslint-config-airbnb": "^15.1.0",
     "eslint-config-airbnb-standard": "^1.4.1",
-    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-import": "^2.13.0",
```
Comment (reactivepixel) [inline package.json:27]: Updating ESLint should be a separate branch. Each feature should be in a single unique pull request allowing for a narrowly focused team discussion.

File: bot/client.js [hunk 2]
```diff
@@ -57,4 +57,15 @@ client.on('message', (message) => {
   }
 });
 
+client.on('guildMemberAdd', (member) => {
+  // Send dm to user
+  member.send(`Welcome to our server ${member}!`);
+  // Send message to general channel
+  const channel = member.guild.channels.find('name', 'general');
+  if (!channel) {
+    return;
```
Comment (reactivepixel) [inline bot/client.js:66]: Erroring silently is not a good scenario. at least kick back to the debugger or craft a better message to respond to the user with, though in this case that would not be appropriate as a user did not kick off this interaction.

File: bot/client.js [hunk 3]
```diff
@@ -57,4 +57,15 @@ client.on('message', (message) => {
   }
 });
 
+client.on('guildMemberAdd', (member) => {
```
Comment (reactivepixel) [inline bot/client.js:60]: Narrowly scoped solution implemented. This is not reusable or extensible in the same ways that the ```client.on('message'``` is. There you can see that multiple controllers are ran against the ```message``` event to allow multiple processings. Currently, If another thing were to need to be implemented on the guildmemberadd event your proposed change would need to be rewritten heavily.

File: bot/controllers/welcome.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        'Resend the welcome message you received from the bot when you first joined the server.',
+        this.resendWelcomeMessage.bind(controller),
+        'dm',
+      ),
+    ];
+  }
+
+  resendWelcomeMessage() {
+    const { message } = this;
+    return `Welcome to our server ${message.member}!`;
```
Comment (reactivepixel) [inline bot/controllers/welcome.js:22]: Duplicate logic: see ```client.js```

File: bot/client.js [hunk 5]
```diff
@@ -57,4 +57,15 @@ client.on('message', (message) => {
...
+  // Send dm to user
+  member.send(`Welcome to our server ${member}!`);
+  // Send message to general channel
+  const channel = member.guild.channels.find('name', 'general');
+  if (!channel) {
+    return;
+  }
+  channel.send(`Welcome to the server ${member}!`);
```
Comment (reactivepixel) [inline bot/client.js:68]: Duplicate logic: see ```welcome.js```

**Pull request closed without merging**